core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -516,7 +516,8 @@ private[deploy] class Worker(
         val cleanupFuture: concurrent.Future[Unit] = concurrent.Future {
           val appDirs = workDir.listFiles()
           if (appDirs == null) {
-            throw new IOException("ERROR: Failed to list files in " + appDirs)
+            throw new IOException(
+              s"ERROR: Failed to list files in ${appDirs.mkString("dirs(", ", ", ")")}")
```

Comment:
@LuciferYang This will cause a NullPointerException error because appDirs is null.

Reply:
My bad, thanks for your correction. Should we print `workdir` here?

Reply:
Yes, let's just print workDir.

Reply:
OK